File: buffer/gpu/src/jvmMain/java/com/wsr/gpu/elementwise/compare/JCompare.java
package com.wsr.gpu.elementwise.compare;

public class JCompare {
    public native void greaterThanD1ToD0(long x, float y, long result, long runtime);
    public native void greaterThanD1ToD1(long x, long y, long result, long runtime);

    public native void lessThanD1ToD0(long x, float y, long result, long runtime);
    public native void lessThanD1ToD1(long x, long y, long result, long runtime);

    public native void equalsD1ToD0(long x, float y, float absoluteTolerance, float relativeTolerance, long result, long runtime);
    public native void equalsD1ToD1(long x, long y, float absoluteTolerance, float relativeTolerance, long result, long runtime);
}

File: buffer/gpu/src/jvmAndroidMain/kotlin/com/wsr/knist/gpu/elementwise/compare/JCompare.kt
package com.wsr.knist.gpu.elementwise.compare

object JCompare {
    external fun greaterThanD1ToD0(x: Long, y: Float, result: Long, runtime: Long)
    external fun greaterThanD1ToD1(x: Long, y: Long, result: Long, runtime: Long)

    external fun lessThanD1ToD0(x: Long, y: Float, result: Long, runtime: Long)
    external fun lessThanD1ToD1(x: Long, y: Long, result: Long, runtime: Long)

    external fun equalsD1ToD0(
        x: Long,
        y: Float,
        absoluteTolerance: Float,
        relativeTolerance: Float,
        result: Long,
        runtime: Long,
    )
    external fun equalsD1ToD1(
        x: Long,
        y: Long,
        absoluteTolerance: Float,
        relativeTolerance: Float,
        result: Long,
        runtime: Long,
    )
}
